src/gfxinit.cpp

```diff
@@ -347,6 +337,17 @@ void GfxLoadSprites()
 	UpdateCursorSize();
 }
 
+GraphicsSet::GraphicsSet()
+	: BaseSet<GraphicsSet, MAX_GFT, true>{}, palette{}, blitter{}
+{
+	// only here due to incomplete types
+}
+
+GraphicsSet::~GraphicsSet()
+{
+	// only here due to incomplete types
+}
```

Comment:
Why are these necessary now? Also, prefer member initialization of `palette` and `blitter` instead of initialization list.

Reply:
The destructor of `std::unique_ptr` needs a complete type, so it can call the destructor of the held type.
I do not want to include `newgrf_config.h` in `base_media_base.h`, so I have to put `~GraphicsSet` into some `.cpp` file.
For some reason MSVC also needs the constructor in a `.cpp` file.

Reply:
Ah, so something like this might be helpful:
`/* Explicit constructor/destructor required due to incomplete type used in unique_ptr type. */`